bionic/flow.py

```diff
@@ -1231,9 +1232,40 @@ def get(self, name, collection=None, fmt=None, mode=object):
             return set(values)
         elif collection is pd.Series or collection == "series":
             if len(result_group.key_space) > 0:
-                index = multi_index_from_case_keys(
-                    case_keys=[result.query.case_key for result in result_group],
-                    ordered_key_names=list(result_group.key_space),
+                ancestor_values_by_case_key_by_name = defaultdict(dict)
+                ancestor_key_space_by_name = {}
+                for ancestor_name in result_group.key_space:
+                    ancestor_result_group = self._deriver.derive(
+                        entity_dnode_from_descriptor(ancestor_name)
+                    )
+                    ancestor_key_space_by_name[
+                        ancestor_name
+                    ] = ancestor_result_group.key_space
+                    for ancestor_result in ancestor_result_group:
+                        ancestor_case_key = ancestor_result.query.case_key
+                        ancestor_values_by_case_key_by_name[ancestor_name][
+                            ancestor_case_key
+                        ] = ancestor_result.value
+
+                orig_case_keys = [result.query.case_key for result in result_group]
+                index = pd.MultiIndex.from_tuples(
+                    tuples=[
+                        tuple(
+                            HashableWrapper(
+                                value=ancestor_values_by_case_key_by_name[
+                                    ancestor_name
+                                ][
+                                    orig_case_key.project(
+                                        ancestor_key_space_by_name[ancestor_name]
+                                    )
+                                ],
+                                token=orig_case_key.tokens[ancestor_name],
+                            )
+                            for ancestor_name in result_group.key_space
+                        )
+                        for orig_case_key in orig_case_keys
+                    ],
+                    names=list(result_group.key_space),
```

Comment:
nit: This code assumes that when you iterate over the elements of `result_group.key_space`, you get them in the same order each time. This happens to be true because `CaseKeySpace` happens to be implemented as a `tuple`, but if it were implemented as a `set` it would not be true. Since a key space is conceptually similar to a set, we might want to pick one ordering and use it each time.